### DISCUSSION TITLE
Computed Properties not working

### DISCUSSION THREAD

**Discussion by Lukmauu** [livewire/livewire#5573] [1. Help] [answered]
Hi Everyone, 

As I understood, Computed Properties were cached, but I have:

``` 
public function getTasksProperty(): \Illuminate\Pagination\LengthAwarePaginator
{
        return $this->owner->tasks()
                ->withCount(['events', 'users'])
                ->paginate(10)
        );
}
```

And on the view, I called ```$this->tasks```,  but to my surprise, this query runs in every little request, It is a table with all tasks, and the user has to select a task to edit because it is set a variable on the component with the Id that is a trip to the server which call this query again.    

This behavior should happen because this query would run only once when the page loads. I also tried to use get() and avoid pagination, but I need help.

**Comment by joshhanley** [comment, marked as answer]
@Lukmauu computed properties aren't cached, they're only memoized within a single request. Any subsequent request will cause the computed property to be "computed" again, thereby running the query again.

I would recommend using `wire:model.defer` as much as possible, which prevents unnecessary network requests from firing and also then minimising the amount of database queries run.

Hope this helps!

**Reply by Lukmauu** [reply]
As I understood, as long I do not make a request, this Tasks data will be available free of charge. Unfortunately, I cannot defer in my situation, but I will use the cache as much as possible. Thank you very much @joshhanley
